runtime/src/main/java/io/micronaut/logging/impl/LogbackLoggingSystem.java

```diff
@@ -35,11 +42,31 @@
 @Internal
 public final class LogbackLoggingSystem implements LoggingSystem {
 
+    private static String DEFAULT_LOGBACK_LOCATION = "logback.xml";
+
+    @Property(name = "logger.config")
+    private Optional<String> logbackXmlLocation;
+
     @Override
     public void setLogLevel(String name, LogLevel level) {
         getLoggerContext().getLogger(name).setLevel(toLevel(level));
     }
 
+    @Override
+    public void refresh() {
+        getLoggerContext().reset();
+        JoranConfigurator joranConfigurator = new JoranConfigurator();
+        joranConfigurator.setContext(getLoggerContext());
+        String logbackXml = logbackXmlLocation.orElse(DEFAULT_LOGBACK_LOCATION);
+
+        try {
+            URL resource = getClass().getClassLoader().getResource(logbackXml);
```

Comment:
@graemerocher  @n0tl3ss , please review this part. I need your opinion on this one

Reply:
I think something like
```
    @Override
    public void refresh() {
        getLoggerContext().reset();
        try {
            new ContextInitializer(getLoggerContext()).autoConfig();
        } catch (JoranException e) {
            throw new LoggingSystemException("Error while refreshing Logback", e);
        }
    }
```

 would do a trick.